cs-utilities/src/main/java/io/cloudslang/content/utilities/services/osdetector/OsDetectorService.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * (c) Copyright 2017 Hewlett-Packard Enterprise Development Company, L.P.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License v2.0 which accompany this distribution.
+ *
+ * The Apache License is available at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+*/
+package io.cloudslang.content.utilities.services.osdetector;
+
+import io.cloudslang.content.utilities.entities.OperatingSystemDetails;
+import io.cloudslang.content.utilities.entities.OsDetectorInputs;
+
+import java.io.IOException;
+
+/**
+ * Created by Tirla Florin-Alin on 28/11/2017.
+ **/
+public interface OsDetectorService {
+    OperatingSystemDetails detect(OsDetectorInputs nmapInputs);
```

Comment:
Maybe it will be more suitable the name 'detectOs'. The child classes will get aligned with the parent class 'OperatingSystemDetector.'